File: 0779-k-th-symbol-in-grammar/0779-k-th-symbol-in-grammar.py
class Solution:
    def kthGrammar(self, n: int, k: int) -> int:
        def solve(n,k):
            if n==1 or k==1:
                return 0
            else:
                if k>2**(n-2):
                    a=solve(n-1,k-2**(n-2))
                    if a==0:
                        return 1
                    else:
                        return 0
                else:
                    return solve(n-1,k)
        return solve(n,k)
    

File: 0779-k-th-symbol-in-grammar/0779-k-th-symbol-in-grammar.cpp
class Solution {
    // int solve(int n,)
public:
    int kthGrammar(int n, int k) {
        if(k==1)return 0;
        else
        {
            // cout<<n<<" "<<k<<endl;
            if(pow(2,(n-2))>=k)
                return kthGrammar(n-1,k);
            return !(kthGrammar(n-1,k-pow(2,(n-2))));
        }    
    }
};
    // 0
    //     01
    //     0110
    //     01101001
    //     3 4